Added transitive inference modelimplementation in folder relational

borukhd opened the pull request.

borukhd committed:
Add files via upload

borukhd committed:
Merge pull request #1 from borukhd/borukhd-patch-3

Add files via upload

borukhd committed:
moved to directory

nriesterer commented:
I encountered a couple of problems while trying to run your code:

1. Could you please update the path specifications in your benchmark JSON files. Currently they specify relative paths `data/empty-TI.csv`. However, the data folder is located in the parent directory. Therefore, it should read: `../data/empty-TI.csv`
2. Even with the fixed paths, I still get the following error:

```
Evaluating '.../cogmods/relational/student_projects/2020_borukhson/models/Baseline/Rational/trueAnswer.py' (1/14)...
Traceback (most recent call last):
  File "/usr/local/bin/ccobra", line 11, in <module>
    load_entry_point('ccobra', 'console_scripts', 'ccobra')()
  File ".../ccobra/ccobra/benchmark/runner.py", line 187, in entry_point
    main(args)
  File ".../ccobra/ccobra/benchmark/runner.py", line 119, in main
    res_df = eva.evaluate()
  File ".../ccobra/ccobra/benchmark/evaluator.py", line 116, in evaluate
    pre_model = importer.instantiate(modelinfo.args)
  File ".../ccobra/ccobra/benchmark/modelimporter.py", line 201, in instantiate
    return self.class_attribute(**model_kwargs)
  File ".../cogmods/relational/student_projects/2020_borukhson/models/Baseline/Rational/trueAnswer.py", line 18, in __init__
    super().__init__(name, ['spatial-relational'], ['single-choice'], commands)
TypeError: __init__() takes 4 positional arguments but 5 were given
```

I assume this is because you used a customised version of CCOBRA. Could you please elaborate briefly on why you had to change CCOBRA for your work?

borukhd committed:
Corrected number of model parameters

borukhd commented:
Hi, 
I corrected the data file path in the JSON files. 
The additional "commands" input value was required for parameter optimization only, it is for this purpose I changed CCOBRA. I removed it, let me know if it works now.

nriesterer commented:
I still get errors:

**trueAnswer.py model**
```
Evaluating '.../cogmods/relational/student_projects/2020_borukhson/models/Baseline/Rational/trueAnswer.py' (1/14)...
Traceback (most recent call last):
  File "/usr/local/bin/ccobra", line 11, in <module>
    load_entry_point('ccobra', 'console_scripts', 'ccobra')()
  File ".../ccobra/ccobra/benchmark/runner.py", line 187, in entry_point
    main(args)
  File ".../ccobra/ccobra/benchmark/runner.py", line 119, in main
    res_df = eva.evaluate()
  File ".../ccobra/ccobra/benchmark/evaluator.py", line 188, in evaluate
    prediction = model.predict(copy.deepcopy(task['item']), **task['aux'])
  File ".../cogmods/relational/student_projects/2020_borukhson/models/Baseline/Rational/trueAnswer.py", line 25, in predict
    return int(self.correctReply((first, second)))
AttributeError: 'CorrectReply' object has no attribute 'correctReply'
```

**benchmarkForHumScrm.json**
```
FileNotFoundError: [Errno 2] File .../cogmods/relational/student_projects/2020_borukhson/data/HumScrm-TI.csv does not exist: '.../cogmods/relational/student_projects/2020_borukhson/data/HumScrm-TI.csv'
```

Additionally, you specified your domain as `spacional-relational`. Is this a typo that should read `spatial-relational`?

Please make sure that all of your benchmarks run without issues.

borukhd committed:
fixed dependencies

borukhd commented:
I reinstalled the ccobra version from this repository and fixed some dependencies, now it should work with every benchmark. Let me know, if there are are any more issues.

nriesterer commented:
It works fine now, thanks. In the following are the results I get on my machine for some exemplary benchmarks of yours. Please check if they look similar to what you got on your computer. I will merge your pull request after your confirmation.

**benchmarkForCamarena.json:**
![Screenshot 2020-05-29 at 15 54 16](https://user-images.githubusercontent.com/2379203/83267549-a9a44b00-a1c4-11ea-83f7-4e53ea3048c7.png)

**benchmarkForHumScrm.json:**
![Screenshot 2020-05-29 at 15 56 22](https://user-images.githubusercontent.com/2379203/83267746-f425c780-a1c4-11ea-8e85-e6cfc7c23ad4.png)

**benchmarkForRGMDL3.json:**
![Screenshot 2020-05-29 at 15 59 13](https://user-images.githubusercontent.com/2379203/83268043-5aaae580-a1c5-11ea-856e-f63be850ed85.png)

borukhd commented:
Thank you. Yes, this matches the results on my computer.

nriesterer commented:
Thank you very much!
